lib/ui/text.dart

```diff
@@ -2231,17 +2300,22 @@ ByteData _encodeStrut(
     bitmask |= 1 << 4;
     data.setFloat32(byteCount, height, _kFakeHostEndian);
     byteCount += 4;
+    if (leadingDistribution != null) {
+      bitmask |= 1 << 5;
+      data.setInt8(byteCount, leadingDistribution.index);
```

Comment:
This does not match the logic in `decodeStrut`.

`decodeStrut` assumes that all the float fields are placed after the byte fields in the buffer.  But `_encodeStrut` is placing the `leadingDistribution` byte after the `height` float.

Also, the `byteCount += 4` here does not match the `setInt8`, and it could exceed the size of the `ByteData` buffer.

I'd recommend moving the encoding of this field up so that it precedes the float fields.